src/Streamly/Internal/Data/Array/Foreign/Mut/Type.hs

```diff
@@ -677,10 +678,15 @@ roundUpLargeArray size =
             ((size + blockSize - 1) .&. negate blockSize)
     else size
 
-{-
 roundUpToPower2 :: Int -> Int
-roundUpToPower2 = undefined
--}
+roundUpToPower2 x =
+    let z = x - 1
+        z1 = z .|. z `shiftR` 1
+        z2 = z1 .|. z1 `shiftR` 2
+        z3 = z2 .|. z2 `shiftR` 4
+        z4 = z3 .|. z3 `shiftR` 8
+        z5 = z4 .|. z4 `shiftR` 16
+        in z5 + 1
```

Comment:
This assumes the word size to be 32-bit. 

We can use `#include "MachDeps.h"` and test `WORD_SIZE_IN_BITS` to be 64 or 32 and handle appropriately. For 64-bit word we will need a z6 as well.